### PR TITLE
EMB-15862: 2-byte documentation.

### PR DESCRIPTION
**Pull request by peter-lasota**
Change to rs232-usb.md to document new extended binary data messages.
Change to protocol.md to document the new flow control.

### CODE REVIEW

File: src/hardware/developing-an-iox/protocol.md [hunk 1]
```diff
@@ -281,6 +281,18 @@ Sent from the IOX to the GO device to inform the GO device of events or status c
 | 0-1 | 0x0004 |
 | 2 | Unused |
 
+#### Information Type 11 IOX Third-Party Tx-Buffer State (GO9+ only)
```
Comment (Stephen-F) [inline src/hardware/developing-an-iox/protocol.md:284]: Suggest changing the wording of GO9+ to (GO9 and newer only). We know what GO9+ is referring to internally, but there is actually a product called GO9+. It is a GO9 with WiFi.

File: src/hardware/developing-an-iox/protocol.md [hunk 2]
```diff
@@ -281,6 +281,18 @@ Sent from the IOX to the GO device to inform the GO device of events or status c
...
+The IOX sends a "Stop-sending" message to GO for every Tx Data(0x0B) message it receives when the IOX Tx-queue going to an external third-party device is about to fill-up.
+When the GO receives the  "Stop-sending" message, it should pause sending messages out from the GO Tx-message queue to the IOX. 
+When the IOX Tx-queue going to an external third-party device returns to the level that it can accept more messages, the IOX will send out a "Resume-sending" message to the GO. 
+When the GO receives the "Resume-sending" message, it will resume sending messages out from the GO Tx-message-queue to the IOX.
+For downloading binary passthrough data to an external third-party device, in the case when the server is sending data faster than the GO can send out to the IOX, the GO Tx-message queue to the IOX will eventually fill-up and data may be dropped.
+
+| Parameter Type | Description |
+| --- | --- |
+| 0-1 | 0x000B |
+| 2 | Status:<br>0 = "Resume-sending": The IOX can accept more messages, the GO can resume sending passthrough messages to the Third-Party device.<br>1 = "Stop-sending": The IOX Tx-queue going to an external third-party device is about to fill-up. |
```
Comment (Stephen-F) [inline src/hardware/developing-an-iox/protocol.md:294]: Remove references to "external third-party device". That's what we use it for, but not necessarily what someone creating a custom IOX would use it for.

File: src/hardware/developing-an-iox/protocol.md [hunk 3]
```diff
@@ -281,6 +281,18 @@ Sent from the IOX to the GO device to inform the GO device of events or status c
 | 0-1 | 0x0004 |
 | 2 | Unused |
 
+#### Information Type 11 IOX Third-Party Tx-Buffer State (GO9+ only)
+The IOX sends a "Stop-sending" message to GO for every Tx Data(0x0B) message it receives when the IOX Tx-queue going to an external third-party device is about to fill-up.
```
Comment (Stephen-F) [inline src/hardware/developing-an-iox/protocol.md:285]: Should change wording to "IOX can choose to send a"...

File: src/hardware/developing-an-iox/protocol.md (hunk 1 above)
Comment (peter-lasota) [reply src/hardware/developing-an-iox/protocol.md]: Fixed.

File: src/hardware/developing-an-iox/protocol.md (hunk 2 above)
Comment (peter-lasota) [reply src/hardware/developing-an-iox/protocol.md]: Reword the descriptions.

File: src/hardware/developing-an-iox/protocol.md (hunk 3 above)
Comment (peter-lasota) [reply src/hardware/developing-an-iox/protocol.md]: Changed as suggested.

File: src/hardware/developing-an-iox/protocol.md [hunk 4]
```diff
@@ -281,6 +281,18 @@ Sent from the IOX to the GO device to inform the GO device of events or status c
 | 0-1 | 0x0004 |
 | 2 | Unused |
 
+#### Information Type 11 - IOX Third-Party Tx-Buffer State (GO9 with WIFI only)
+The IOX can choose to send a "Stop-sending" message to GO for every Tx Data(0x0B) message it receives when the IOX believes its internal buffer is about to be overrun.<br>
+When the GO receives the  "Stop-sending" message, it should pause sending messages out from the GO Tx-message queue to the IOX.<br>
+When the IOX internal buffer returns to the level that it can accept more messages, the IOX will send out a "Resume-sending" message to the GO.<br>
```
Comment (Stephen-F) [inline src/hardware/developing-an-iox/protocol.md:287]: "IOX will send out", suggest changing to "IOX should send out".

File: src/hardware/developing-an-iox/protocol.md [hunk 5]
```diff
@@ -281,6 +281,18 @@ Sent from the IOX to the GO device to inform the GO device of events or status c
...
+When the GO receives the "Resume-sending" message, it will resume sending messages out from the GO Tx-message-queue to the IOX.<br>
+For downloading binary passthrough data, in the case when the server is sending data faster than the GO can send out to the IOX, the GO Tx-message queue to the IOX will eventually fill-up and data may be dropped.
+
+| Parameter Type | Description |
+| --- | --- |
+| 0-1 | 0x000B |
+| 2 | Status:<br>0 = "Resume-sending": The IOX can accept more messages, the GO can resume sending passthrough messages to the Third-Party device.<br>1 = "Stop-sending": The IOX Tx-queue is about to fill-up. |
```
Comment (Stephen-F) [inline src/hardware/developing-an-iox/protocol.md:294]: "messages to the Third-Party device", suggest changing to "messages to the IOX"

File: src/hardware/developing-an-iox/protocol.md (hunk 1 above)
Comment (peter-lasota) [reply src/hardware/developing-an-iox/protocol.md]: Renamed the information type to: 

> Information Type 11 - IOX Stop/Resume GO Tx messages (GO9 and newer only)

File: src/hardware/developing-an-iox/protocol.md (hunk 4 above)
Comment (peter-lasota) [reply src/hardware/developing-an-iox/protocol.md]: Fixed.

File: src/hardware/developing-an-iox/protocol.md (hunk 5 above)
Comment (peter-lasota) [reply src/hardware/developing-an-iox/protocol.md]: Fixed.

File: src/hardware/developing-an-iox/protocol.md [hunk 6]
```diff
@@ -281,6 +281,18 @@ Sent from the IOX to the GO device to inform the GO device of events or status c
 | 0-1 | 0x0004 |
 | 2 | Unused |
 
+#### Information Type 11 - IOX Stop/Resume GO Tx messages (GO9 and newer only)
+The IOX can choose to send a "Stop-sending" message to GO for every Tx Data(0x0B) message it receives when the IOX believes its internal buffer is about to be overrun.<br>
+When the GO receives the  "Stop-sending" message, it should pause sending messages out from the GO Tx-message queue to the IOX.<br>
```
Comment (Stephen-F) [inline src/hardware/developing-an-iox/protocol.md:286]: Looks like there are 2 spaces before "Stop-sending"

File: src/hardware/developing-an-iox/protocol.md [hunk 7]
```diff
@@ -283,15 +283,15 @@ Sent from the IOX to the GO device to inform the GO device of events or status c
...
+When the GO receives the "Stop-sending" message, it should pause sending messages out from the GO Tx-message queue to the IOX.<br>
 When the IOX internal buffer returns to the level that it can accept more messages, the IOX should send out a "Resume-sending" message to the GO.<br>
 When the GO receives the "Resume-sending" message, it will resume sending messages out from the GO Tx-message-queue to the IOX.<br>
 For downloading binary passthrough data, in the case when the server is sending data faster than the GO can send out to the IOX, the GO Tx-message queue to the IOX will eventually fill-up and data may be dropped.
 
 | Parameter Type | Description |
 | --- | --- |
 | 0-1 | 0x000B |
-| 2 | Status:<br>0 = "Resume-sending": The IOX can accept more messages, the GO can resume sending passthrough messages to the IOX.<br>1 = "Stop-sending": The IOX Tx-queue is about to fill-up. |
+| 2 | Status:<br>0 = "Resume-sending": The IOX can accept more messages, the GO can resume sending passthrough messages to the IOX.<br>1 = "Stop-sending": The IOX interal buffer is about out to be overun. |
```
Comment (Stephen-F) [inline src/hardware/developing-an-iox/protocol.md:294]: "The IOX interal buffer is about out to be overun", should be: "The IOX internal buffer is about to overrun".

**Pull request closed without merging**